src/cmake/thirdparty/FindOpenCASCADE.cmake

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+# other Axom Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: (BSD-3-Clause)
+#------------------------------------------------------------------------------
+# Setup OpenCASCADE
+#------------------------------------------------------------------------------
+# This file defines:
+#  OPENCASCADE_FOUND - If OpenCASCADE was found
+#  OPENCASCADE_INCLUDE_DIRS - The OpenCASCADE include directories
+#  OPENCASCADE_LIBRARIES - The OpenCASCADE libraries
+#------------------------------------------------------------------------------
+
+# First check for OPENCASCADE_DIR
+if (NOT EXISTS "${OPENCASCADE_DIR}")
+    message(FATAL_ERROR "Given OPENCASCADE_DIR does not exist: ${OPENCASCADE_DIR}")
+endif()
+
+if (NOT IS_DIRECTORY "${OPENCASCADE_DIR}")
+    message(FATAL_ERROR "Given OPENCASCADE_DIR is not a directory: ${OPENCASCADE_DIR}")
+endif()
```

Comment:
```suggestion
axom_assert_is_directory(DIR_VARIABLE OPENCASCADE_DIR)
```

Reply:
Thanks -- I changed it.

Looking at the pattern from our other `FindFoo.cmake` files, we check that `FOO_DIR` is a valid directory (using the `axom_assert_is_directory` macro) before we include `FindFoo.cmake`, but then we also check that `FOO_DIR` is defined at the top of the file.

See, e.g:
* https://github.com/LLNL/axom/blob/e5b5707ed09eeb46265b5740a810242970eaaf76/src/cmake/thirdparty/SetupAxomThirdParty.cmake#L275-L280
* https://github.com/LLNL/axom/blob/e5b5707ed09eeb46265b5740a810242970eaaf76/src/cmake/thirdparty/FindLUA.cmake#L11-L18

and also
* https://github.com/LLNL/axom/blob/e5b5707ed09eeb46265b5740a810242970eaaf76/src/cmake/thirdparty/SetupAxomThirdParty.cmake#L296-L301
* https://github.com/LLNL/axom/blob/e5b5707ed09eeb46265b5740a810242970eaaf76/src/cmake/thirdparty/FindC2C.cmake#L14-L21

Should we clean these up (in a follow-up PR) or leave them as is?

Reply:
whatever makes you the happiest @kennyweiss